embrace-android-delivery/src/test/kotlin/io/embrace/android/embracesdk/internal/delivery/execution/HttpUrlConnectionRequestExecutionServiceTest.kt

```diff
@@ -0,0 +1,204 @@
+package io.embrace.android.embracesdk.internal.delivery.execution
+
+import io.embrace.android.embracesdk.internal.comms.api.ApiResponse
+import io.embrace.android.embracesdk.internal.delivery.SupportedEnvelopeType
+import okhttp3.Headers.Companion.toHeaders
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.mockwebserver.SocketPolicy
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import java.net.UnknownHostException
+
+class HttpUrlConnectionRequestExecutionServiceTest {
```

Comment:
Might want to test timeout as well?

Reply:
I am testing timeout, but it's a bit hidden 😅 

```
    fun `return incomplete if the server returns an unexpected response code`() {
        // given a server that disconnects before sending a response
        server.enqueue(MockResponse().setSocketPolicy(SocketPolicy.DISCONNECT_AT_START))
        ...
```

As the socket disconnects when it receives a request, it will not return anything, so the actual error is a timeout. I'll add that as a comment